### PR TITLE
CCP-181 Support for json

### PR DESCRIPTION
**Pull request by mkarthik90**

### CODE REVIEW

**Comment by jdamerow** [comment]
damn sorry, looks like I never submitted that review.

**Comment by mkarthik90** [comment]
@jdamerow  Addressed review comments. I will resolve merge conflicts once CCP-176 gets merged to develop.

File: Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/json/ConceptSuperTypeJson.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
+package edu.asu.conceptpower.servlet.json;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+public class ConceptSuperTypeJson {
```
Comment (jdamerow) [inline Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/json/ConceptSuperTypeJson.java]: remove

File: Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/json/JsonConceptMessage.java [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+
+public class JsonConceptMessage implements IConceptMessage {
+
+    private URIHelper uriCreator;
+
+    public JsonConceptMessage(URIHelper uriCreator) {
+        this.uriCreator = uriCreator;
+    }
+
+    public String getAllConceptMessage(Map<ConceptEntry, ConceptType> entries) {
```
Comment (jdamerow) [inline Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/json/JsonConceptMessage.java]: ?

File: Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/json/JsonConceptMessage.java [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+            conceptMessages.add(getConceptMessage(entry, entries.get(entry)));
+        }
+        
+        String json = null;
+        
+        try{
+            json = mapper.writeValueAsString(conceptMessages);
+        } catch(JsonProcessingException ex) {
+            System.out.println("Supressing error");
+            ex.printStackTrace();
```
Comment (jdamerow) [inline Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/json/JsonConceptMessage.java]: baahhhhhhhhh

File: Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/json/JsonConceptMessage.java [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+        json.setDescription(StringEscapeUtils.escapeXml10(entry.getDescription()));
+        json.setConceptList(StringEscapeUtils.escapeXml10(entry.getConceptList()));
+
+        if (entry.getChangeEvents() != null && !entry.getChangeEvents().isEmpty()) {
+            List<ChangeEvent> changeEvents = entry.getChangeEvents();
+            Collections.sort(changeEvents);
+            json.setCreatorId(StringEscapeUtils.escapeXml10(changeEvents.get(0).getUserName() != null ? changeEvents.get(0).getUserName().trim() : ""));
+        } else {
+            json.setCreatorId(
+                    StringEscapeUtils.escapeXml10(entry.getCreatorId() != null ? entry.getCreatorId().trim() : ""));
```
Comment (jdamerow) [inline Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/json/JsonConceptMessage.java]: check if jackson escapes on its own

File: Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/json/JsonTypeMessage.java [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+    private URIHelper uriCreator;
+
+    public JsonTypeMessage(URIHelper uriCreator) {
+        this.uriCreator = uriCreator;
+    }
+
+    @Override
+    public String getConceptTypeMessage(ConceptType type, ConceptType supertype) {
+
+        ObjectMapper mapper = new ObjectMapper();
```
Comment (jdamerow) [inline Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/json/JsonTypeMessage.java]: define where used

File: Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/json/JsonTypeMessage.java [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+            superTypeJson.setTypeName(StringEscapeUtils.escapeXml10(supertype.getTypeName()));
+            jsonType.setSuperType(superTypeJson);
+        }
+
+        String json = null;
+        try {
+            json = mapper.writeValueAsString(jsonType);
+        } catch (JsonProcessingException ex) {
+            // TODO
+            ex.printStackTrace();
```
Comment (jdamerow) [inline Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/json/JsonTypeMessage.java]: bahhhhhhhh!!!!!

File: Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/rest/ConceptIDLookup.java [hunk 7]
```diff
@@ -84,35 +84,38 @@
...
-                    entry = conceptManager.getConceptEntry(entry.getId());
-                }
-                addAlternativeIds(pathParts[lastIndex], entry);
+                // Check if the id used in a generic id. If so fetch the
+                // concept wrapper id for the generic wordnet id
+                entry = checkAndAddAlternativeIds(pathParts, lastIndex, entry);
             } catch (LuceneException e) {
                 logger.error("Lucene Exception", e);
-                return new ResponseEntity<String>(msg.getXML(xmlEntries), HttpStatus.INTERNAL_SERVER_ERROR);
+                return new ResponseEntity<String>(msg.getAllConceptMessage(entryMap), HttpStatus.INTERNAL_SERVER_ERROR);
```
Comment (jdamerow) [inline Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/rest/ConceptIDLookup.java]: This should return an error message.

File: Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/rest/ConceptLookup.java [hunk 8]
```diff
@@ -73,11 +74,24 @@
...
         }
-        
+
+        Map<ConceptEntry, ConceptType> entryMap = generateEntryMap(entries);
+        IConceptMessage conceptMessage = messageFactory.getMessageFactory(acceptHeader).createConceptMessage();
+
+        String xmlEntries = null;
+        if (entries != null) {
+            xmlEntries = conceptMessage.getAllConceptMessage(entryMap);
+
```
Comment (jdamerow) [inline Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/rest/ConceptLookup.java]: remove

File: Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/rest/ConceptSearch.java [hunk 9]
```diff
@@ -89,19 +91,22 @@
...
 
-        List<String> xmlEntries = new ArrayList<String>();
         Map<ConceptEntry, ConceptType> entryMap = new HashMap<ConceptEntry, ConceptType>();
 
-        XMLConceptMessage msg = messageFactory.createXMLConceptMessage();
+        IConceptMessage msg = messageFactory.getMessageFactory(acceptHeader).createConceptMessage();
+        createEntryMap(searchResults, entryMap);
+
+        return new ResponseEntity<String>(msg.getAllConceptMessage(entryMap), HttpStatus.OK);
+
```
Comment (jdamerow) [inline Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/rest/ConceptSearch.java]: remove

File: Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/rest/SynonymSearch.java [hunk 10]
```diff
@@ -84,23 +86,23 @@
...
+            entries = msg.getAllConceptMessage(entryMap);
 		}
 
 		HttpHeaders responseHeaders = new HttpHeaders();
         responseHeaders.add("Content-Type", "text/html; charset=utf-8");
         
-        return new ResponseEntity<String>(msg.getXML(xmlEntries), responseHeaders, HttpStatus.OK);
+        return new ResponseEntity<String>(entries, responseHeaders, HttpStatus.OK);
 	}
 }
```
Comment (jdamerow) [inline Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/rest/SynonymSearch.java]: formatting

File: Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/rest/TypeIdLookup.java [hunk 11]
```diff
@@ -87,6 +88,6 @@
 		HttpHeaders responseHeaders = new HttpHeaders();
         responseHeaders.add("Content-Type", "text/html; charset=utf-8");
         
-        return new ResponseEntity<String>(msg.getXML(xmlEntry), responseHeaders, HttpStatus.OK);
+        return new ResponseEntity<String>(entry, responseHeaders, HttpStatus.OK);
 	}
```
Comment (jdamerow) [inline Conceptpower+Spring/src/main/java/edu/asu/conceptpower/servlet/rest/TypeIdLookup.java]: formatting

**Comment by mkarthik90** [comment]
Due to merge conflicts. All changes from CCP-181 is moved to branch: story/CCP-181-2. From now on all changes will be made at story/CCP-181-2 and PR will be created from CCP-181-2.

**Pull request closed without merging**